dataframely/_base_collection.py

```diff
@@ -188,11 +191,17 @@ def _get_metadata(source: dict[str, Any]) -> Metadata:
         result = Metadata()
 
         # Get all members via the annotations
-        if "__annotations__" in source:
-            for attr, kls in source["__annotations__"].items():
-                result.members[attr] = CollectionMeta._derive_member_info(
-                    attr, kls, CollectionMember()
-                )
+        annotations = {}
+        if sys.version_info >= (3, 14):
+            if "__annotate_func__" in source:
+                annotations = source["__annotate_func__"](Format.VALUE)
```

Comment:
I'm not sure if this is worse than before but it feels a little whacky. I didn't figure out how to use `annotationlib.get_annotations` here though, I don't think that works within the metaclass. Maybe there's something to re-architecture at some point...